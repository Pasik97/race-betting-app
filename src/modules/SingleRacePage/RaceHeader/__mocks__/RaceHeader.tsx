const mock = () => '(RaceHeader /)';

export default mock;
