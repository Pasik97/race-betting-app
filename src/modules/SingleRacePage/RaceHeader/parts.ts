import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const RaceTitleWrapper = styled.div`
   display: flex;
   align-items: baseline;
   justify-content: space-between;
`;

export const RaceTitle = styled.h2`
   color: #01263A;
   font-weight: 400;
   font-size: 24px;
   line-height: 27px;
   margin: 16px 0 0 0;
`;

export const RaceTitleStatus = styled.p`
   margin: 0 0 0 8px;
   font-size: 16px; 
   line-height: 18px;
   white-space: nowrap;
`;

export const StyledLink = styled(Link)`
   display: inline-flex;
   align-items: center;
   height: 32px;
   color: #1a1a1a;
   font-size: 13px;
   line-height: 16px;
   text-decoration: none;
   background: #f5f5f5;
   padding: 0px 20px;
   border-radius: 24px;

   &:hover {
      background: #eeeeee;
   }
`;
