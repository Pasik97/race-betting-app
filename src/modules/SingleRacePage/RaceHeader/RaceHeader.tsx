import React from 'react';
import { useSelector } from 'react-redux';
import * as P from './parts';
import ApplicationState from 'store/ApplicationState';
import { getRaceById } from 'store/races/selectors';

interface RaceHeaderProps {
   raceId: string;
}

const RaceHeader: React.FC<RaceHeaderProps> = (({ raceId }) => {
   const race = useSelector((state: ApplicationState) => getRaceById(state, raceId));

   return (
      <>
         <P.StyledLink to="/races">â€¹ Back to races page</P.StyledLink>
         <P.RaceTitleWrapper>
            {race?.name && <P.RaceTitle title={race.name}>{race.name}</P.RaceTitle>}
            {race?.active !== undefined && <P.RaceTitleStatus>Status: {race.active ? 'Active' : 'Inactive'}</P.RaceTitleStatus>}
         </P.RaceTitleWrapper>
      </>
   )
});

export default RaceHeader;
