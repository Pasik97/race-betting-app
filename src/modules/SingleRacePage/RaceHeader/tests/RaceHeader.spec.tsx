import { create } from "react-test-renderer";
import { Provider } from "react-redux";
import RaceHeader from "../RaceHeader";
import { createMockRaceStore } from "modules/SingleRacePage/tests/mockData";
import { mockRace, mockRacesState, secondMockRace } from "store/races/tests/mockData";
import { BrowserRouter } from "react-router-dom";

describe('<RaceHeader />', () => {
   it('should render link to races page, race name and label "Active" if race is active', () => {
      const component = create(
         <BrowserRouter>
            <Provider store={createMockRaceStore(mockRacesState)}>
               <RaceHeader raceId={mockRace.id.toString()} />
            </Provider>
         </BrowserRouter>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });

   it('should render link to races page, race name and label "Inactive" if race is inactive', () => {
      const component = create(
         <BrowserRouter>
            <Provider store={createMockRaceStore(mockRacesState)}>
               <RaceHeader raceId={secondMockRace.id.toString()} />
            </Provider>
         </BrowserRouter>,
      );

      expect(component.toJSON()).toMatchSnapshot();
   });
});
