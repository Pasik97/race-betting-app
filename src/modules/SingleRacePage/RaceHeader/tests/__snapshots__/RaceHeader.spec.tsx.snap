// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RaceHeader /> should render link to races page, race name and label "Active" if race is active 1`] = `
Array [
  <a
    className="sc-hKwDye jqqwCx"
    href="/races"
    onClick={[Function]}
  >
    ‹ Back to races page
  </a>,
  <div
    className="sc-bdvvtL dfGmNp"
  >
    <h2
      className="sc-gsDKAQ eSsmNL"
      title="mock race"
    >
      mock race
    </h2>
    <p
      className="sc-dkPtRN cVLRqC"
    >
      Status: 
      Active
    </p>
  </div>,
]
`;

exports[`<RaceHeader /> should render link to races page, race name and label "Inactive" if race is inactive 1`] = `
Array [
  <a
    className="sc-hKwDye jqqwCx"
    href="/races"
    onClick={[Function]}
  >
    ‹ Back to races page
  </a>,
  <div
    className="sc-bdvvtL dfGmNp"
  >
    <h2
      className="sc-gsDKAQ eSsmNL"
      title="second mock race"
    >
      second mock race
    </h2>
    <p
      className="sc-dkPtRN cVLRqC"
    >
      Status: 
      Inactive
    </p>
  </div>,
]
`;
